libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -0,0 +1,195 @@
+        initializeViewModels()
+        this.lifeCycleOwner = context as? LifecycleOwner ?: throw LifeCycleOwnerNotFoundException()
+        this.lifeCycleOwner.lifecycle.addObserver(this)
+        this.mapView = MapView(context, mapInitOptions)
+    }
+
+    /**
+     * Creates references to view components. Views provided via the [viewProvider] will be given preference.
+     * If null, default Mapbox designed views will be used.
+     */
+    fun configure(viewProvider: ViewProvider) {
+        // add views
+        binding.mapContainer.addView(mapView)
+        maneuverView = viewProvider.maneuverProvider?.invoke() ?: MapboxManeuverView(context)
+        binding.maneuverContainer.addView(maneuverView)
+        speedLimitView = viewProvider.speedLimitProvider?.invoke() ?: MapboxSpeedLimitView(context)
+        binding.speedLimitContainer.addView(speedLimitView)
+        tripProgressView =
+            viewProvider.tripProgressProvider?.invoke() ?: MapboxTripProgressView(context)
+        binding.speedLimitContainer.addView(tripProgressView)
+        soundButtonView = viewProvider.soundButtonProvider?.invoke() ?: MapboxSoundButton(context)
+        binding.volumeContainer.addView(soundButtonView)
+        recenterButtonView =
+            viewProvider.recenterButtonProvider?.invoke() ?: MapboxRecenterButton(context)
+        binding.recenterContainer.addView(recenterButtonView)
+        routeOverviewButtonView =
+            viewProvider.routeOverviewButtonProvider?.invoke() ?: MapboxRouteOverviewButton(context)
+        binding.routeOverviewContainer.addView(routeOverviewButtonView)
+
+        renderStateMutations()
+        performActions()
+
+        update(navigationViewOptions)
+    }
+
+    fun update(navigationViewOptions: NavigationViewOptions) {
+        this.navigationViewOptions = navigationViewOptions
+        // todo trigger relayout
+    }
+
+    fun retrieveMapView(): MapView = mapView
+
+    fun retrieveMapboxNavigation(): MapboxNavigation = mapboxNavigation
+
+    private fun initializeViewModels() {
+        this.navigationStateViewModel =
+            ViewModelProvider(activity).get(NavigationStateViewModel::class.java)
+        this.containerVisibilityViewModel =
+            ViewModelProvider(activity).get(ContainerVisibilityViewModel::class.java)
+    }
+
+    private fun performActions() {
+        lifeCycleOwner.lifecycleScope.launch {
+            navigationStateViewModel.processAction(
+                flowOf(
+                    NavigationStateAction.ToEmpty
+                )
+            )
+            containerVisibilityViewModel.processAction(
+                flowOf(
+                    ContainerVisibilityAction.ForEmpty
+                )
+            )
+        }
+    }
+
+    private fun renderStateMutations() {
+        lifeCycleOwner.lifecycleScope.launch {
+            navigationStateViewModel.navigationState().collect { _ ->
+            }
```

Comment:
Can you hook up this part? Reason I'm saying this, is because how are you connecting the NavigationState flow to the containerVisibilityViewModel state flow?

Reply:
I am not connecting the `NavigationState` to container visibility just yet, because we still don't know what needs to happen based on each of these states. I would leave it blank for now but would add more code once we have a clear understanding.